app/precompiled.py

```diff
@@ -196,16 +197,22 @@ def sanitise_file_contents(encoded_string, *, allow_international_letters, filen
             message = "letter-too-long"
             raise ValidationFailed(message, page_count=page_count)
 
-        message, invalid_pages = get_invalid_pages_with_message(file_data)
+        message, invalid_pages = get_invalid_pages_with_message(
+            file_data, is_an_attachment=is_an_attachment
+        )
         if message:
             raise ValidationFailed(message, invalid_pages, page_count=page_count)
 
-        file_data, recipient_address = rewrite_pdf(
-            file_data,
-            page_count=page_count,
-            allow_international_letters=allow_international_letters,
-            filename=filename,
-        )
+        if is_an_attachment:
+            file_data = normalise_fonts_and_colours(file_data, filename)
+            recipient_address = None
+        else:
+            file_data, recipient_address = rewrite_pdf(
```

Comment:
Did you consider passing `is_an_attachment` into this function and decide not to do it? It feels like we are having to pull out some of that function's internals into this function now - which is maybe a bit less clean?

I can imagine `rewrite_pdf` getting a bit messy if we pass that flag in - could we maybe add a `rewrite_attachment_pdf` with the same interface and just the operations we want? It would keep both of those functions implementations simple, and keep their internal decisions about what rewriting needs to take place out of this function.

Reply:
Yeah, I considered passing `is_an_attachment` to `rewrite_pdf`, but I decided it would be clunky because of the order of operations (where attachments need only the middle two operations).

I also quite like how it got divided: `normalise_fonts_and_colours` deals with general preparation of the PDF for printing, while `rewrite_pdf` does operations related to more DVLA-specific requirements.

Reply:
If I rename `normalise_fonts_and_colours` to `rewrite_subsequent_pages`, would that work better?

Reply:
Hmmm ... I'm not sure about that, because we'd be calling `rewrite_subsequent_pages` on _all_ of the PDF's pages if the document is an attachment.

Reply:
but the attachment on the whole is subsequent pages of a letter.

Reply:
Still, I quite like as it is now, so happy to leave it ☺️

Reply:
Yes, I realise. We're just debating semantics really. I'm just letting you know I still find it confusing - you can rethink it or not :)